website/content/docs/upgrading/raft-wal.mdx

```diff
@@ -0,0 +1,52 @@
+---
+layout: docs
+page_title: Upgrading integrated storage to raft-wal
+description: These are instructions for how to upgrade integrated storage to use the raft-wal log store.
+---
+
+# Upgrading integrated storage to use raft-wal
+
+## What is it?
+
+This document will cover how to upgrade your Vault cluster that's using integrated storage with the raft-boltdb
+library, which has been the default since integrated storage's inception, to use the new raft-wal log store
+introduced in 1.16.0. It will also cover how to revert an upgrade of raft-wal back to using raft-boltdb, if
+desired. This document will not cover what raft-wal is or why you might want to upgrade to it.
+For further information on those types of questions, please refer to the
+[internals](/vault/docs/internals/integrated-storage#configurable-raft-log-store) page.
+
+## How to upgrade from raft-boltdb to raft-wal
+
+Migrating from raft-boltdb to raft-wal can be done on a node-by-node basis. The steps to perform on each node are:
+1. Gracefully shut down the Vault process.
+1. Rename the existing Vault storage directory to something else.
+   1. For example, if the `path` in Vault's `storage` config is
+   `/opt/vault/data` then you'd do something like: `mv /opt/vault/data /opt/vault/data.old`
+1. Add the new config
+   1. Add `raft_wal = "true"` to the existing storage stanza in the Vault config file.
+   1. Optionally enable log verification by setting `raft_log_verifier_enabled = "true"`.
+   1. If log verification was enabled, optionally choose a verification interval, if the default of one minute is not optimal.
+      1. This can be set by adding e.g. `raft_log_verification_interval = "30s"` to the storage stanza.
+      1. Choose your own interval instead of `30s` as demonstrated here. Note that `10s` is the minimum allowed value.
```

Comment:
HInted at before but have we tested that this is safe on a mixed cluster? If not we should call out with a warning that you must not enable verification until all nodes int he cluster are running 1.16. i.e. don't do this _during_ the upgrade to 1.16 or a new node might become leader early and crash all the old followers.